test/IISIntegration.FunctionalTests/Inprocess/IISTests.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Server.IntegrationTesting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Testing;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests
+{
+    public class IISTests : LoggedTest
+    {
+        public IISTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        [SkipIfIISExpressSchemaMissingInProcess]
+        [SkipIfNotAllowedToRunIIS]
+        public Task HelloWorld_IISExpress_Clr_X64_Standalone()
+        {
+            return HelloWorld(RuntimeFlavor.CoreClr, ApplicationType.Standalone);
+        }
+
+        [Fact]
+        [SkipIfIISExpressSchemaMissingInProcess]
+        [SkipIfNotAllowedToRunIIS]
+        public Task HelloWorld_IISExpress_Clr_X64_Portable()
+        {
+            return HelloWorld(RuntimeFlavor.CoreClr, ApplicationType.Portable);
+        }
+
+        private async Task HelloWorld(RuntimeFlavor runtimeFlavor, ApplicationType applicationType)
+        {
+            var testName = $"HelloWorld_{runtimeFlavor}";
+            using (StartLog(out var loggerFactory, testName))
+            {
+                var logger = loggerFactory.CreateLogger("HelloWorldTest");
+
+                var deploymentParameters = Helpers.GetBaseDeploymentParameters();
+                deploymentParameters.ApplicationType = applicationType;
+
+                using (var deployer = new IISDeployer(deploymentParameters, loggerFactory))
+                {
+                    var deploymentRes = await deployer.DeployAsync();
+                    var deploymentResult = new IISDeploymentResult(deploymentRes, logger);
+
+                    if (applicationType == ApplicationType.Standalone)
+                    {
+                        Helpers.ModifyAspNetCoreSectionInWebConfig(deploymentResult, "processPath", "InProcessWebSite.exe");
```

Comment:
Shouldn't this go into deployer?

Reply:
Yeah. Lot of this will go away once we move this into hosting.

Reply:
Why not now?

Reply:
We may not move it into hosting.

Reply:
Let's try keeping in here and see if it's easier to maintain. However you'd need to add a separate package if you want it to be consumed in other repos. That can wait though.

Reply:
Alright, I'll err on the side of putting everything in here then.